xain_fl/grpc/test_coordinator.py

```diff
@@ -196,3 +234,42 @@ def test_remove_participant():
     coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant1")
 
     assert coordinator.state == coordinator_pb2.State.ROUND
+
+
+def test_number_of_selected_participants():
+    # test that the coordinator needs minimum 3 participants and selects 2 of them
+    coordinator = Coordinator(
+        minimum_participants_in_round=2, fraction_of_participants=0.6
+    )
+    coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant1")
+    coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant2")
+
+    # the coordinator should not start a round but wait for a third
+    assert coordinator.minimum_connected_participants == 3
+    assert coordinator.state == coordinator_pb2.State.STANDBY
+
+    coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant3")
+
+    # now the coordinator must have started a round and selected 2 participants
+    assert coordinator.state == coordinator_pb2.State.ROUND
+    assert len(coordinator.round.participant_ids) == 2
```

Comment:
and what does `coordinator.round` look like prior to this?

Reply:
good question. so `coordinator.round` is initialised with the list of `Participants.ids()` (plural Participants, defined inside `grpc.coordinator` module), which at the time of initialisation is an empty list. Then when time of selection comes (and only then) `coordinator.round` is updated inside `Coordinator.select_participant_ids_and_init_round()` with the list of selected ids. so `coordinator.round.participant_ids` should be empty until the selection happens. 

i have update the test to make this more clear.